src/filesystem_helper.cpp

```diff
@@ -306,6 +306,22 @@ path temp_directory_path()
   return path(temp_path);
 }
 
+path create_temp_directory(const std::string & base_name, const path & parent)
+{
+  auto template_path = base_name + ".XXXXXX";
```

Comment:
this only guarantees not to return errno `EINVAL`, but `mkdtemp` possibly has errno from `mkdir`. we can refer to man page, it tells that `Also see mkdir(2) for other possible values for errno.` e.g) what if the user application does not have access permission to that parent directory? in that case, `EACCES` will be return with null pointer.